Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -27,9 +63,82 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Frontend' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Widget' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+            'Enlight_Controller_Action_PreDispatch_Frontend_Checkout' => 'writeDefaultPaymentToSession',
+            'Shopware_Modules_Admin_UpdatePayment_FilterSql' => 'onApplePayButtonChangePaymentMethod',
         ];
     }
 
+    public function writeDefaultPaymentToSession(\Enlight_Event_EventArgs $args)
+    {
+        if (
```

Comment:
is this something like 
if ($isDefaultPaymentAllowed) ?
then please class + unit test, thx

Reply:
Restructured. Unit test not added yet.